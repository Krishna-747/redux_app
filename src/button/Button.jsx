import React, { useState } from "react";
import styles from "./Buttonstyles.module.css";

function Button({ btn_val }) {
  const [state, setState] = useState(0);
  const [value, setValue] = useState("");

  const [ digit, setDigit ] = useState({
      arr: [0, 1, 2, 3, 4, 5, 6, 7, 8, 9],
      operation: { add: '+', subtract: '-', multiply: 'X', equals: '=' }
  })

  const txtbutton = (num, add) => {
    setState(state + num + add);
  };

  const handleChange = (e) => {
    setValue(e.target.value);
  };

  const calButton = (e) => {
    const clickedvalues = e.target.innerHTML
    

  }

  const valButton = (e) => {
    console.log(e.target.innerHTML)
  }

  return (
    <>
      <input
        data-testid="input_val"
        type="text"
        onChange={handleChange}
        value={value}
      />
      <div data-testid="stateid">{state}</div>
      <button
        data-testid="button"
        onClick={() => txtbutton(10, 20)}
        className={styles.btnstyle}
      >
        So {btn_val}
      </button>
      <button>Added button</button>
      {
          digit.arr.map((elm, key) => <button onClick={valButton} key={key}>
              {elm}
          </button>)
      }
      <button onClick={calButton}>{digit.operation.add}</button>
      <button onClick={calButton}>{digit.operation.subtract}</button>
      <button onClick={calButton}>{digit.operation.multiply}</button>
      <button onClick={calButton}>{digit.operation.equals}</button>
    </>
  );
}

export default Button;
