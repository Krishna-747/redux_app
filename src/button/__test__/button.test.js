import { render, fireEvent } from '@testing-library/react'
import Button from '../../button/Button'
import renderer from 'react-test-renderer'

describe('Main Branch', () => {
    it('react-test', () => {
        render(<Button></Button>)
    })

    it('react-test-button 1', () => {
        const { getByTestId } = render(<Button btn_val='Get'></Button>)
        expect(getByTestId('button')).toHaveTextContent('Get')
    })

    it('react-test-button 2', () => {
        const { getByTestId } = render(<Button btn_val='Save'></Button>)
        expect(getByTestId('button')).toHaveTextContent('Save')
    })

    it('react-snapshot', () => {
        const tree = renderer.create(<Button></Button>).toJSON()
        expect(tree).toMatchSnapshot()
    })

    it('click button', () => {
        const { getByTestId } = render(<Button></Button>)
        const btnclick = getByTestId('button')
        const btn = getByTestId('stateid')
        const mat = Number(btn.innerHTML)
        expect(mat).toBe(0)
        fireEvent.click(btnclick)
        const sat = Number(btn.innerHTML)
        expect(sat).toBe(30)
    })

    it('input button', () => {
        const { getByTestId } = render(<Button></Button>)
        const search = getByTestId('input_val')
        fireEvent.change(search, { target: { value: 'First Text' } })
        expect(search.value).toBe('First Text')
    })
})