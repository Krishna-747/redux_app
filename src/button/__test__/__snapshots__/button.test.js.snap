// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Branch react-snapshot 1`] = `
Array [
  <input
    data-testid="input_val"
    onChange={[Function]}
    type="text"
    value=""
  />,
  <div
    data-testid="stateid"
  >
    0
  </div>,
  <button
    className="btnstyle"
    data-testid="button"
    onClick={[Function]}
  >
    So 
  </button>,
  <button>
    Added button
  </button>,
  <button
    onClick={[Function]}
  >
    0
  </button>,
  <button
    onClick={[Function]}
  >
    1
  </button>,
  <button
    onClick={[Function]}
  >
    2
  </button>,
  <button
    onClick={[Function]}
  >
    3
  </button>,
  <button
    onClick={[Function]}
  >
    4
  </button>,
  <button
    onClick={[Function]}
  >
    5
  </button>,
  <button
    onClick={[Function]}
  >
    6
  </button>,
  <button
    onClick={[Function]}
  >
    7
  </button>,
  <button
    onClick={[Function]}
  >
    8
  </button>,
  <button
    onClick={[Function]}
  >
    9
  </button>,
  <button
    onClick={[Function]}
  >
    +
  </button>,
  <button
    onClick={[Function]}
  >
    -
  </button>,
  <button
    onClick={[Function]}
  >
    X
  </button>,
  <button
    onClick={[Function]}
  >
    =
  </button>,
]
`;
